tests/unit/test_extensionPoints.py

```diff
@@ -48,56 +48,259 @@ def test_isWeak(self):
+			def handlerMethod(self, a, **kwargs):
+				calledKwargs['a'] = a
+				calledKwargs.update(kwargs)
+
+		h = handlerClass()
+		extensionPoints.callWithSupportedKwargs(h.handlerMethod, a='a value', b='b value')
+		self.assertEqual(calledKwargs, {'a': 'a value', 'b': 'b value'})
+
+	def test_memberHandlerTakesParamsAndKwargs_givenPositional(self):
+		"""Test to ensure that a member handler gets the correct arguments, including implicit "self"
+		Handler takes parameter and kwargs.
+		callWithSupportedKwargs given a positional.
+		Handler should get positional.
+		"""
+		calledKwargs = {}
+
+		class handlerClass():
+			def handlerMethod(self, a, **kwargs):
+				calledKwargs['a'] = a
+				calledKwargs.update(kwargs)
+
+		h = handlerClass()
+		extensionPoints.callWithSupportedKwargs(h.handlerMethod, 'a value')
+		self.assertEqual(calledKwargs, {'a': 'a value'})
+
+	def test_memberHandlerTakesParams_givenPositional(self):
+		"""Test to ensure that a member handler gets the correct arguments, including implicit "self"
+		Handler takes parameter.
+		callWithSupportedKwargs given a positional.
+		Handler should get positional.
+		"""
+		calledKwargs = {}
+
+		class handlerClass():
+			def handlerMethod(self, a):
+				calledKwargs['a'] = a
 
-	def test_supportsNoKwargs(self):
+		h = handlerClass()
+		extensionPoints.callWithSupportedKwargs(h.handlerMethod, 'a value')
+		self.assertEqual(calledKwargs, {'a': 'a value'})
+
+	def test_memberHandlerTakesParams_givenKwargs(self):
+		"""Test to ensure that a member handler gets the correct arguments, including implicit "self"
+		Handler takes parameter.
+		callWithSupportedKwargs given a keyword arg.
+		Handler should get kwarg.
+		"""
+		calledKwargs = {}
+
+		class handlerClass():
+			def handlerMethod(self, a):
+				calledKwargs['a'] = a
+
+		h = handlerClass()
+		extensionPoints.callWithSupportedKwargs(h.handlerMethod, a='a value')
+		self.assertEqual(calledKwargs, {'a': 'a value'})
+
+	def test_handlerTakesNoParams_noKwargsGiven_handlerIsCalled(self):
+		"""Essentially another veration of `test_handlerParamsMatchKwargsGiven_valuePassedIn` test
+		The handler function expects no params.
+		callWithSupportedKwargs is given no kwargs.
+		Expectation: handler is still called.
+		"""
 		called = []
 		def handler():
 			called.append(True)
-		extensionPoints.callWithSupportedKwargs(handler, a=1)
+		extensionPoints.callWithSupportedKwargs(handler)
 		self.assertEqual(called, [True])
 
-	def test_supportsLessKwargs(self):
-		gotKwargs = {}
-		def handler(a=None):
-			gotKwargs["a"] = a
-		extensionPoints.callWithSupportedKwargs(handler, a=1, b=2)
-		self.assertEqual(gotKwargs, {"a": 1})
-
-	def test_supportsExtraKwargs(self):
-		gotKwargs = {}
-		def handler(a=None, b=2):
-			gotKwargs["a"] = a
-			gotKwargs["b"] = b
-		extensionPoints.callWithSupportedKwargs(handler, a=1)
-		self.assertEqual(gotKwargs, {"a": 1, "b": 2})
+	def test_handlerTakesNoParams_someKwargsGiven_handlerIsCalled(self):

```

Comment:
`someKwargsGiven` seems to break the pattern here (and a few other places below); previously, you have the words `takes` and `given` first. I know "some kwargs given" is more linguistically correct than "given some kwargs", but I wonder whether consistency might win out here.